src/Avalonia.Themes.Default/ToggleSwitch.xaml

```diff
@@ -0,0 +1,188 @@
+            </Grid>
+          </ControlTemplate>
+        </Setter>
+
+      </Style>
+       
+        
+      <Style Selector="ToggleSwitch:unchecked /template/ Ellipse#PART_CircleThumb">
+        <Style.Animations>
+          <Animation Duration="0:0:0.4" FillMode="Forward" Easing="CubicEaseOut">
+            <KeyFrame Cue="0%">
+              <Setter Property="Fill" Value="White"/>
+              <Setter Property="Canvas.Left" Value="28"/>
+            </KeyFrame>
+            <KeyFrame Cue="100%">
+              <Setter Property="Fill" Value="RoyalBlue"/>
+              <Setter Property="Canvas.Left" Value="5"/>
+            </KeyFrame>
+          </Animation>
+        </Style.Animations>
+      </Style>
+      
+    <Style Selector="ToggleSwitch:checked /template/ Ellipse#PART_CircleThumb">
+        <Style.Animations>
+          <Animation Duration="0:0:0.4" FillMode="Forward" Easing="CubicEaseOut">
+            <KeyFrame Cue="0%">
+              <Setter Property="Fill" Value="RoyalBlue"/>
+              <Setter Property="Canvas.Left" Value="5"/>
+            </KeyFrame>
+            <KeyFrame Cue="100%">
+              <Setter Property="Fill" Value="White"/>
+              <Setter Property="Canvas.Left" Value="28"/>
+            </KeyFrame>
+          </Animation>
+        </Style.Animations>
+      </Style>
+  
+      <Style Selector="ToggleSwitch:unchecked /template/ Border#PART_BorderBackground">
+        <Style.Animations>
+          <Animation Duration="0:0:0.5" FillMode="Forward" Easing="CubicEaseOut">
+            <KeyFrame Cue="0%">
+              <Setter Property="Background" Value="Royalblue"/>
+            </KeyFrame>
+            <KeyFrame Cue="100%">
+              <Setter Property="Background" Value="White"/>
+            </KeyFrame>
+          </Animation>
+        </Style.Animations>
+        <Setter Property="BorderBrush" Value="Silver"/>
+      </Style>
+
+      <Style Selector="ToggleSwitch:checked /template/ Border#PART_BorderBackground">
+        <Style.Animations>
+          <Animation Duration="0:0:0.5" FillMode="Forward" Easing="CubicEaseOut">
+             <KeyFrame Cue="0%">
+              <Setter Property="Background" Value="White"/>
+            </KeyFrame>
+            <KeyFrame Cue="100%">
+              <Setter Property="Background" Value="RoyalBlue"/>
+            </KeyFrame>
+          </Animation>
+        </Style.Animations>
+        <Setter Property="BorderBrush" Value="Royalblue"/>
+      </Style>
+
+   <Style Selector="ToggleSwitch:pressed  /template/ Border#PART_BorderBackground">
+    <Setter Property="BorderBrush" Value="{DynamicResource ThemeControlHighBrush}"/>
+  </Style>
+ 
+      <Style Selector="ToggleSwitch:checked /template/ ContentControl#PART_TextBooleanValue">
+        <Setter Property="Content" Value="On"/>
+      </Style>
+
+      <Style Selector="ToggleSwitch:unchecked /template/ ContentControl#PART_TextBooleanValue">
+        <Setter Property="Content" Value="Off"/>
```

Comment:
I think, Off and On values should be moved to property from the start?
https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.toggleswitch.offcontent?view=winrt-19041
https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.toggleswitch.oncontent?view=winrt-19041

Reply:
@maxkatz6 Yes, in the future the control wil evolve to a more flexibel control.

Reply:
![Toggle](https://user-images.githubusercontent.com/58340108/84761013-1d1dc900-afc9-11ea-8721-202c7f004666.gif)